tidb-performance-tuning-config.md

```diff
@@ -0,0 +1,527 @@
+```
+
+### Tune coprocessor cache for read-heavy workloads
+
+For read-heavy workloads, optimizing the coprocessor cache can significantly improve query performance. The [coprocessor cache](https://docs.pingcap.com/tidb/stable/coprocessor-cache) stores the results of coprocessor requests, reducing the need to recompute frequently accessed data. To optimize cache performance:
+
+1. Monitor the current cache hit ratio using the metrics described in the [coprocessor cache documentation](https://docs.pingcap.com/tidb/stable/coprocessor-cache)
+2. Consider increasing the cache size for better hit rates on larger working sets
+3. Adjust the admission threshold based on your query patterns
+
+Here are the recommended settings for a read-heavy workload:
+
+```toml
+[tikv-client.copr-cache]
+capacity-mb = 4096
+admission-max-ranges = 5000
+admission-max-result-mb = 10
+admission-min-process-ms = 0
+```
+
+### Optimize chunk size for workload characteristics
+
+The [`tidb_max_chunk_size`](system-variables.md#tidb_max_chunk_size) controls how many rows TiDB processes in a single iteration during query execution. Optimizing this value based on your workload can significantly impact performance:
+
+#### Workload-Specific Recommendations
+
+- **OLTP Workloads** (high concurrency, small transactions):
+    - Recommended range: 128-256 rows (default is 1024)
+    - Benefits: Reduced memory overhead, faster limit query processing
+    - Use case: Point queries, small range scans
+
+    ```sql
+    SET GLOBAL tidb_max_chunk_size = 128;
+    ```
+
+- **OLAP/Analytical Workloads** (complex queries, large result sets):
+    - Recommended range: 1024-4096 rows
+    - Benefits: Improved throughput for large scans
+    - Use case: Aggregations, large table scans
+
+    ```sql
+    SET GLOBAL tidb_max_chunk_size = 4096;
+    ```
+
+### Optimize transaction mode and DML type for different workloads
+
+TiDB provides different transaction modes and DML execution types to optimize performance for various workload patterns:
+
+#### Transaction Modes
+
+- **Pessimistic Mode** (Default):
+    - Best for general workloads with potential write conflicts
+    - Provides better consistency guarantees
+
+    ```sql
+    SET SESSION tidb_txn_mode = "pessimistic";
+    ```
+
+- **Optimistic Mode**:
+    - Suitable for workloads with minimal write conflicts
+    - Better performance for multi-statement transactions
+    - Example: `begin; insert...; insert...; commit;`
+
+    ```sql
+    SET SESSION tidb_txn_mode = "optimistic";
+    ```
+
+#### DML Types
+
+`tidb_dml_type = "bulk"` indicates the bulk DML execution mode, which is suitable for scenarios where a large amount of data is written, causing excessive memory usage in TiDB.
+
+- **Pipeline DML** (New in v8.0.0):
+    - Ideal for bulk data loading with no conflicts
+    - Reduces TiDB memory usage during large writes
+    - Requirements:
+        - Auto-commit must be enabled
+        - `pessimistic-auto-commit` must be false
```

Comment:
```suggestion
To use the bulk DML execution mode, set `tidb_dml_type` to `"bulk"`. This mode optimizes bulk data loading without conflicts and reduces memory usage during large write operations. Before using this mode, ensure that:

- Auto-commit is enabled.
- The [`pessimistic-auto-commit`](/tidb-configuration-file.md#pessimistic-auto-commit-new-in-v600) configuration item is set to `false`.
```
Pipelined DML is the underlying implementation that enables Bulk DML. For consistency with our documentation, we should continue using the term "Bulk DML" when referring to this feature.